bindings/java/src/main/com/apple/foundationdb/tuple/FastByteComparisons.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * ByteArrayUtil.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb.tuple;
+
+import java.lang.reflect.Field;
+import java.nio.ByteOrder;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Comparator;
+
+import sun.misc.Unsafe;
+
+
+/**
+ * Utility code to do optimized byte-array comparison.
+ * This is borrowed and slightly modified from Guava's {@link UnsignedBytes}
+ * class to be able to compare arrays that start at non-zero offsets.
+ */
+abstract class FastByteComparisons {
+
+    private static final int UNSIGNED_MASK = 0xFF;
+
+    /**
+     * Lexicographically compare two byte arrays.
+     */
+    public static int compareTo(byte[] b1, int s1, int l1, byte[] b2, int s2, int l2) {
```

Comment:
Would you mind using more descriptive names for these arguments? If I understand correctly, this is the function being called elsewhere in the code, so it's probably helpful to have descriptive names.

Reply:
It looks like this function also has some expected preconditions that we may want to document. In particular:

0 <= s1, s2, l1, l2
s1+l1 <= b1.length
s2+l2 <= b2.length
b1, b2 != null